master/internal/restore.go

```diff
@@ -182,11 +182,7 @@ func (e *experiment) restoreTrial(
 	if trialID != nil {
 		t.id = *trialID
 		t.idSet = true
-		if _, ok := e.searcher.TrialsCreated[searcher.Create.RequestID]; !ok {
-			ctx.Tell(ctx.Self(), trialCreated{
-				requestID: searcher.Create.RequestID,
-			})
-		}
+		t.trialCreationSent = e.searcher.TrialsCreated[searcher.Create.RequestID]
```

Comment:
If I'm understanding this change right now we now will always send ```trialCreated``` at ```buildTaskSpec``` time where before in restore case we could send it earlier sometimes? This makes more sense to me.

Just thinking out loud here.